### ISSUE TITLE
License

### ISSUE THREAD

**Issue by bwright86** [mattheyan/PsData#1]
Hello maththeyan,

I was wanting to use a copy of your module in my workplace, but I do not see any licensing on the module, allowing or preventing commercial use.

I do see a "All rights reserved" copyright that is generated from New-ModuleManifest. But this doesn't indicate rights to copy or share.

Could you update your copyright to something else? (i.e. MIT License, GPL, or any of these: https://opensource.org/licenses)

I have tried the module out, and like how well it works to import/export PSData files.

I would appreciate it,
Brent

**Comment by mattheyan** [comment]
Hi Brent,

This is from a blog post by @Jaykul from a while back. Unfortunately it can no longer be found on his blog, but I was able to recover it via way back machine, or something like that (see the mht file in the repository).

I didn't see any kind of license or notes about rights attached to the original blog post, so I haven't attached any license for that reason, just tried to attribute the original source in any way I could. If he's OK with MIT-ing it, I certainly wouldn't mind.  ;)

Thanks,
Bryan

**Comment by Jaykul** [comment]
That code is now my [Configuration](https://github.com/poshcode/configuration) module, and has a lot more functionality ;-)

**Comment by mattheyan** [comment]
@Jaykul: That's great! I'll look to use that in the future.

@bwright86: Do you think you'd be able to use the [new module](https://github.com/poshcode/configuration) instead? It appears to be MIT licensed.

**Comment by bwright86** [comment]
@Jaykul and @mattheyan Thank you for the quick response! I will look into the module, from a quick look it is pretty full featured.

It appears to store PSData in user profiles, app data and program data. My needs were for a .psd1 importer, that read in files from a path.

I will check the Configuration module out tomorrow and see if it fills my needs. Otherwise I may rewrite it for what I need and pass along credit.

Thanks again for the replies :-)
-Brent

**Comment by mattheyan** [comment]
@bwright86, it looks like `Metadata.pms1`, in that repo, may be what you need.

@Jaykul, just out of curiosity, why did you end up going with the "Metadata" terminology, rather than "PSData"?

Thanks,
Bryan

**Comment by Jaykul** [comment]
Yeah, the core functions in the `Metadata.psm1` will do what you're looking for @bwright86 -- and those commands are, of course, available as exports from Configuration. Just `Install-Module Configuration` and focus on the `Metadata` noun. 

I've resisted separating them because I care more about the configuration scenario, so that's my justification for working on this stuff...

@mattheyan I'm not really sure. PowerShell just refers to it at "LocalizedData" or "DATA" or "StringData" but when I was writing this I was fully in the _configuration_ mode, and metadata made some sense to me in that context.

I think if I was going to separate that module out and release it on it's own, I'd probably just call it `PowerShellData` (v1) since I think that's the meaning of the `.psd1` extension -- and it conveys the powerful extensible nature of this better. I never liked `PSON` as PowerShell Object Notation (as a play on JSON) 😉

**Comment by mattheyan** [comment]
@Jaykul - That makes sense. There doesn't seem to be a lot of known scenarios for using `.psd1` files, other than the module manifest.

To date I've been using this PSData code for ***reading*** simple `.psd1` files, but I may have a need to read and write `.psd1` files in the future beyond what this module can do. If you're open to it, I might want to explore the `PowerShellData` idea. I may ping you about it then to see if there's a way we could do it without putting extra work on your plate.  ;)  If not I could probably just use your `Metadata` module as-is. Either way, its good to know that code found a more permanent home.  :)

@bwright86 - Let me know if you find the `Metadata` module to suite your needs so we can close this out. Thanks.

**Comment by bwright86** [comment]
@mattheyan you are right, I had only looked at the Configuration.psm1 file. The Metadata.psm1 would do what I need, and adds a feature I was going to build (updating an existing .psd1). I also 2nd your idea on splitting ''Metadata'' into the 'PowerShellData' type module.

To give you an idea of what I am using this for. I am building a config generator using templates made up of .psd1 files, most properties are metadata. The last property is a Here-String with the actual data containing tokens that will be replaced with values. The input is a .csv files. Each row has values that will replace the tokens. So it is a big find/replace engine, and the .psd1 format lets me make decisions on which templates will be used, and how they will be folded together.

I chose .psd1 files since they are more human readable, and closer to JSON, than Export-CliXML.

@Jaykul this is a great module, and should be quite useful. I am glad as well that this continued to be developed to it's full potential.

I am happy with the conclusion of this issue, thanks for all of your work to resolve this.
